src/iccpd/src/mlacp_sync_update.c

```diff
@@ -181,164 +246,282 @@ int mlacp_fsm_update_mac_entry_from_peer( struct CSM* csm, struct mLACPMACData *
+
+                            // free only if not in change list to be send to peer node,
+                            // else free is taken care after sending the update to peer
+                            if (!MAC_IN_MSG_LIST(&(MLACP(csm).mac_msg_list), mac_msg, tail))
+                            {
+                                free(mac_msg);
+                            }
+
+                            ICCPD_LOG_ERR(__FUNCTION__, "Ignore Recv MAC ADD "
+                                "MAC %s vlan %d interface %s peer link not available ",
+                                mac_addr_to_str(mac_msg->mac_addr), mac_msg->vid, mac_msg->ifname);
+                            return 0;
+                        }
                     }
                 }
+                else
+                {
+                    /*Update local item*/
+                    memcpy(&mac_msg->ifname, MacData->ifname, MAX_L_PORT_NAME);
+
+                    /*from MCLAG port and the local port is up, add mac to ASIC to update port*/
+                    add_mac_to_chip(mac_msg, mac_msg->fdb_type);
+                }
             }
+	    else if(!from_mclag_intf && (strcmp(mac_msg->ifname, MacData->ifname) == 0))
+            {
+                // local to remote MAC move on Orphan port.
+                if (strlen(csm->peer_itf_name) != 0)
+                {
+                    if (strcmp(mac_msg->ifname, csm->peer_itf_name) == 0)
+                    {
+                        /*This MAC is already point to peer-link*/
+                        ICCPD_LOG_DEBUG("ICCP_FDB", "Remote MAC ADD learn on Orphan port ,MAC already points to Peer_link"
+                            " interface  %s, MAC %s vlan-id %d ", mac_msg->ifname,
+                            mac_addr_to_str(mac_msg->mac_addr), mac_msg->vid);
+                        return 0;
+                    }
 
-            break;
+                    if (csm->peer_link_if && csm->peer_link_if->state == PORT_STATE_UP)
+                    {
+                        /*Redirect the mac to peer-link*/
+                        memcpy(&mac_msg->ifname, csm->peer_itf_name, MAX_L_PORT_NAME);
+
+                        ICCPD_LOG_DEBUG("ICCP_FDB", "Remote MAC ADD learn on Orphan port ,point MAC address to Peer_link"
+                            "interface  %s, MAC %s vlan-id %d ", mac_msg->ifname,
+                            mac_addr_to_str(mac_msg->mac_addr), mac_msg->vid);
+                        /*Send mac add message to mclagsyncd*/
+                        add_mac_to_chip(mac_msg, mac_msg->fdb_type);
+                    }
+                    else
+                    {
+                        /*Redirect the mac to peer-link*/
+                         /*must redirect but if peerlink is down FdbOrch will delete MAC */
+                        memcpy(&mac_msg->ifname, csm->peer_itf_name, MAX_L_PORT_NAME);
+                        add_mac_to_chip(mac_msg, mac_msg->fdb_type);
```

Comment:
@Praveen-Brcm 
You changed the behavior from del_mac_from_chip() to add_mac_to_chip() even when the peerlink is down. Is it intended? Why?
Because the behavior is different from the behavior at line#531, the mac address is added when the peerlink is up.
Thank you!